app/models/travel_expense_report.rb

```diff
@@ -0,0 +1,24 @@
+class TravelExpenseReport < ActiveRecord::Base
+  belongs_to :user
+  has_many :travel_expense_report_items
+  accepts_nested_attributes_for :travel_expense_report_items
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :general_advance, numericality: true
+  validate 'start_before_end_date'
+  validate 'general_advance_positive'
```

Comment:
validates_numericality_of :general_advance_positive, :only_integer => true, :greater_than_or_equal_to => 0
should do the trick in one line ;)